python/ray/serve/_private/http_proxy.py

```diff
@@ -199,12 +199,20 @@ def match_route(
         return None
 
 
-class HTTPProxy:
-    """This class is meant to be instantiated and run by an ASGI HTTP server.
-
-    >>> import uvicorn
-    >>> controller_name = ... # doctest: +SKIP
-    >>> uvicorn.run(HTTPProxy(controller_name)) # doctest: +SKIP
+class GenericProxy:
```

Comment:
mark this as an abstract base class: https://docs.python.org/3/library/abc.html

and methods that need to be implement should be abstract methods

also, any method that needs to be implemented should not be underscore-prefixed